src/pages/loading-waiting-states/progress-bar/index.tsx

```diff
@@ -0,0 +1,234 @@
+        };
+    }, [list]);
+
+    return (
+        <div style={{ minHeight: '100vh' }}>
+            <AppBar data-cy="pxb-toolbar" position={'sticky'}>
+                <Toolbar classes={{ gutters: classes.toolbarGutters }}>
+                    <Hidden mdUp={true}>
+                        <IconButton
+                            data-cy="toolbar-menu"
+                            color={'inherit'}
+                            onClick={(): void => {
+                                dispatch({ type: TOGGLE_DRAWER, payload: true });
+                            }}
+                            edge={'start'}
+                            style={{ marginRight: 20 }}
+                        >
+                            <MenuIcon />
+                        </IconButton>
+                    </Hidden>
+                    <Typography variant={'h6'} color={'inherit'}>
+                        Progress Bars
+                    </Typography>
+                    <div />
+                </Toolbar>
+            </AppBar>
+            <div className={classes.container}>
+                <div className={classes.uploadButtonContainer}>
+                    <Button
+                        variant={'contained'}
+                        color={'primary'}
+                        startIcon={<VerticalAlignTop />}
+                        onClick={uploadFile}
+                    >
+                        <Typography noWrap color={'inherit'}>
+                            UPLOAD NEW FILE
+                        </Typography>
+                    </Button>
+                </div>
+                <Card>
+                    <FormControl className={classes.formControlStyle} component="fieldset">
+                        <RadioGroup aria-label="folder" name="folder" value="1">
+                            {foldersList.map((option, i) => (
+                                <FormControlLabel
+                                    className={classes.formLabel}
+                                    key={i}
+                                    value={option.value}
+                                    control={<Radio />}
+                                    label={
+                                        <div className={classes.radioLabel}>
+                                            <Folder className={classes.iconStyle} />
+                                            <Typography> {option.label} </Typography>
+                                        </div>
+                                    }
+                                />
+                            ))}
+                        </RadioGroup>
+                    </FormControl>
+                </Card>
+                <List data-cy={'list-content'} disablePadding component="nav" className={classes.placementOfList}>
+                    {list.map(
+                        (item, i): JSX.Element => (
+                            <div key={`itemKey${i}`} className={classes.infoListItemStyle}>
```

Comment:
Can you try using [`<Snackbar />`](https://material-ui.com/components/snackbars/)? It would come with the desired animation

Reply:
Also, the way this is implemented right now makes your snackbar progressbar go backward when a old snackbar is removed from `list`. Might want to assign a different `key` prop to different list item.

FYI, `key`s doesn't have to be `string`. You may just do `key={item.id}`

Reply:
> Also, the way this is implemented right now makes your snackbar progressbar go backward when a old snackbar is removed from list.

I suspect this is related to what we talked about before about doing the list manipulation based on index rather than on id. When the list items are added and removed, the indexes get shifted and so you're updating a different item than you intend to.

Reply:
> Can you try using <Snackbar />? It would come with the desired animation

The snackbar component is a bit limited. I know the `message` prop takes a Node, so you could get a two-liner of text in there (you'd still have to override the colors), but not sure about the progress bar...you'd probably have to fight it with styles to make it look right. But you could give it a shot. It might be interesting to see if you could leverage the autoHideDuration (switching it from null to XXX seconds to trigger the dismissal after however many seconds).

Reply:
auto hide, animation that will squeeze another snackbar up when there are multiple snackbars on the screen, automatic themed style (not 100% on this), shadows, fade out animation when the snackbars are dismissed.

All these are the great things about Snackbar. Unless you are determined to reinvent some wheels...

Reply:
I wasn't knocking the Snackbar...if we can get it to work, that would be ideal. I was just commenting on some places where it will probably give some challenges.